src/tree/identity_tree.rs

```diff
@@ -0,0 +1,612 @@
+use std::collections::{BTreeMap, HashMap, VecDeque};
+
+use eyre::{eyre, OptionExt};
+use semaphore::dynamic_merkle_tree::DynamicMerkleTree;
+use semaphore::merkle_tree::{Branch, Hasher};
+use semaphore::poseidon_tree::{PoseidonHash, Proof};
+use semaphore::Field;
+use serde::Serialize;
+
+use super::Hash;
+
+pub enum LeafUpdates {
+    Insert(Leaves),
+    Delete(Leaves),
+}
+
+impl From<LeafUpdates> for Leaves {
+    fn from(val: LeafUpdates) -> Self {
+        match val {
+            LeafUpdates::Insert(leaves) => leaves,
+            LeafUpdates::Delete(leaves) => leaves,
+        }
+    }
+}
+
+// Node index to hash, 0 indexed from the root
+pub type StorageUpdates = HashMap<u32, Hash>;
+
+// Leaf index to hash, 0 indexed from the initial leaf
+pub type Leaves = HashMap<u32, Hash>;
```

Comment:
Consider introducing newtypes:
```
pub struct NodeIndex(pub u32);
pub struct LeafIndex(pub u32);
```

and then discard the type aliases. That way you get type safety and fewer indirections

Reply:
Maybe something like https://github.com/worldcoin/contract-deployer/blob/main/src/types.rs